tests/base/hdf5_07.cc

```diff
@@ -0,0 +1,198 @@
+                                          4, // second point
+                                          5,
+                                          4, // third point
+                                          26,
+                                          12}; // fourth point
+    std::vector<Number>  data_b        = {9, 4, 7, 6};
+
+    std::vector<hsize_t> coordinates_c = {5,
+                                          7, // first point
+                                          1,
+                                          8, // second point
+                                          5,
+                                          5, // third point
+                                          27,
+                                          12, // fourth point
+                                          28,
+                                          29}; // fifth point
+    std::vector<Number>  data_c        = {2, 3, 5, 6, 3};
+
+    if (Utilities::MPI::this_mpi_process(mpi_communicator) == 0)
+      {
+        dataset.write_selection(data_a, coordinates_a);
+      }
+    else if (Utilities::MPI::this_mpi_process(mpi_communicator) == 1)
+      {
+        dataset.write_selection(data_b, coordinates_b);
+      }
+    else if (Utilities::MPI::this_mpi_process(mpi_communicator) == 2)
+      {
+        dataset.write_selection(data_c, coordinates_c);
+      }
+    else
+      {
+        dataset.template write_none<Number>();
+      }
+    deallog << "Sum " + dataset_name << " " << container_name << "<"
+            << type_name << ">"
+            << " (Write): "
+            << std::accumulate(data_a.begin(), data_a.end(), 0) +
+                 std::accumulate(data_a.begin(), data_a.end(), 0) +
+                 std::accumulate(data_a.begin(), data_a.end(), 0)
+            << std::endl;
+  }
+}
+
+
+
+int
+main(int argc, char **argv)
+{
+  initlog();
+
+  try
+    {
+      Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
+
+      MPI_Comm mpi_communicator = MPI_COMM_WORLD;
+
+      ConditionalOStream pcout(
+        std::cout, Utilities::MPI::this_mpi_process(mpi_communicator) == 0);
+
+      std::string                filename           = "test.h5";
+      const std::vector<hsize_t> dataset_dimensions = {50, 30};
+
+      HDF5::File data_file(filename,
+                           HDF5::File::FileAccessMode::create,
+                           mpi_communicator);
+
+      write_test<std::vector, double>(data_file,
+                                      dataset_dimensions,
+                                      mpi_communicator,
+                                      pcout);
+
+      AssertThrow(Utilities::MPI::this_mpi_process(mpi_communicator) == 0,
+                  ExcInternalError());
+
+      write_test<FullMatrix, double>(data_file,
+                                     dataset_dimensions,
+                                     mpi_communicator,
+                                     pcout);
```

Comment:
Interestingly the test hangs with the following code
```
write_test<FullMatrix, double>(data_file, dataset_dimensions, mpi_communicator, pcout);
```
But the test does not hang with the following code
```
write_test<std::vector, double>(data_file, dataset_dimensions, mpi_communicator, pcout);
```